src/socket.rs

```diff
@@ -827,23 +827,27 @@ impl Type {
         Type(c::SOCK_DGRAM)
     }
 
+    #[cfg(not(target_os = "redox"))]
```

Comment:
This is a public API and so I'd prefer to not change the public interface of this crate depending on the platform currently. Could the redox module use a sentinel value for this constant which causes a runtime error when used in this crate?

Reply:
Good point! Updated :)